kafka-0.10.x/src/main/scala/monix/kafka/KafkaConsumerObservableManualCommit.scala

```diff
@@ -42,24 +42,26 @@ final class KafkaConsumerObservableManualCommit[K, V] private[kafka] (
   // Caching value to save CPU cycles
   private val pollTimeoutMillis = config.fetchMaxWaitTime.toMillis
 
+  case class CommitWithConsumer(consumer: KafkaConsumer[K, V]) extends Commit {
```

Comment:
Any specific reason to have a class for it?

Reply:
`CommitWithConsumer` is a case class so the objects of it will be equal when they have same consumer. If just new `Commit` objects won't.